hack/generator/pkg/astmodel/package_definition.go

```diff
@@ -23,19 +23,20 @@ type PackageDefinition struct {
 	PackageName      string
 	GeneratorVersion string
 	definitions      Types
+	fileWriter       *FileWriter
 }
 
 // NewPackageDefinition constructs a new package definition
 func NewPackageDefinition(groupName string, packageName string, genVersion string) *PackageDefinition {
-	return &PackageDefinition{groupName, packageName, genVersion, make(Types)}
+	return &PackageDefinition{groupName, packageName, genVersion, make(Types), NewFileWriter()}
```

Comment:
We almost want a way for each file type to return the right kind of file writer. Maybe?

This looks as though you're assuming a package may only contain AST files - is that right? What if we want to create documentation or YAML files that go along as part of the package?

Reply:
I think that in this case the package is the go construct, so it seems reasonable to think that the contents of the package are valid go code?

I'm not sure if reality bears that out in my head things such as YAML files, etc aren't part of the package really - they might be in the folder but they're not part of the logical code-package?

Reply:
Heh. Different definitions. To me, the package is the directory - any files in the directory that aren't Go code had better still be related to what the package does; if not, they should be moved.

Reply:
Yeah I see how you got there. I'm not actually sure the technical meaning of `package` in the go context - is it just the code in the folder or everything? For example if I import an package that has some YAML files in it do the YAML files come along with the import or just the .go files?

In any case though I think in our context today the `PackageDefinition` is purely dealing with Go files and isn't capable of emitting non-go files (since `definitions` is the "set of things to generate" and it is just a collection of `Types`). If/when we add support for other types of generations we can revisit this and include a type->writer mapping?